nitro-protocol/contracts/NitroAdjudicator.sol

```diff
@@ -101,20 +92,19 @@ contract NitroAdjudicator is ForceMove, MultiAssetHolder {
     }
 
     /**
-    * @notice Check that the submitted pair of states form a valid transition (public wrapper for internal function _requireValidTransition)
-    * @dev Check that the submitted pair of states form a valid transition (public wrapper for internal function _requireValidTransition)
-    * @param nParticipants Number of participants in the channel.
-    transition
-    * @param ab Variable parts of each of the pair of states
-    * @param appDefinition Address of deployed contract containing application-specific validTransition function.
-    * @return true if the later state is a validTransition from its predecessor, reverts otherwise.
-    */
-    function validTransition(
-        uint256 nParticipants,
-        IForceMoveApp.VariablePart[2] memory ab, // [a,b]
-        address appDefinition
-    ) public pure returns (bool) {
-        return _requireValidTransition(nParticipants, ab, appDefinition);
+     * @notice Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @dev Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @param fixedPart Fixed part of the state channel.
+     * @param signedVariableParts Array of variable parts to find the latest of.
+     * @return VariablePart Latest supported by application variable part from supplied array.
+     */    
+    function latestSupportedState(
+        FixedPart calldata fixedPart,
+        SignedVariablePart[] calldata signedVariableParts
+    ) external pure returns (VariablePart memory) {
+        // copy to memory explicitly to avoid `Stack to deep` error
```

Comment:
This could also have an issue to track it? I guess it is a TODO to return to this at some point?

Reply:
Using `memory` instead of `calldata` adds up to 100 gas to `conclude`, but `NitroAdjudicator` deployment consumed 100k less gas.

This made me think about compiler optimization runs and the question arose: How was number "200" for optimization runs chosen?
I have reverted to using `calldata`, changed runs to "1000" and gas savings on functions are 500-800, but deployment consumes 50k more.

I believe, gas costs on functions are more important than on deployment, thus it would be logical to increase optimization runs and use `calldata` with internal copying to `memory` (as presented in commented piece of code).

What do you think, @geoknee ?

Reply:
I think 200 is just the default value, it wasn't chosen by us for any particular reason. 

Yes I agree that deployment costs are less important than function calls. So your proposed approach makes sense.

Reply:
So then I will change the comment to something more precise and not related to TODO.
Re optimization runs: I believe it is not related to this PR at all, thus will be put in a separate issue and tackled once this is merged.